src/utilities/layer-from-json.ts

```diff
@@ -0,0 +1,17 @@
+/* eslint-disable prettier/prettier */
+import { layerTypes } from '../layer';
+import { ILayer, ILayerJSON } from '../layer/base-layer';
+
+export default function layerFromJSON(
+  jsonLayer: ILayerJSON,
+  ...args: unknown[]
```

Comment:
I think args here refer to as `ILayerSettings`. More info: https://github.com/BrainJS/brain.js/blob/master/src/layer/base-layer.ts#L27

Reply:
Sorry - correct me if I'm wrong - but doesn't some of the layers have `ILayerSettings` and `ILayer` as args.

Eg. (from `src\layer\activation.ts`)
```typescript
  constructor(inputLayer: ILayer, settings?: ILayerSettings) {
```

Reply:
I think also `inputLayer` is supplied by here: https://github.com/BrainJS/brain.js/blob/f572a3d2489950cf530d7f7cee7721167744d351/src/feed-forward.ts#L697

In `layerFromJSON()`'s args.

Reply:
Lets focus on getting types for layer constructor, then this will be easier to handle. https://github.com/BrainJS/brain.js/pull/611#discussion_r499871647